include/jemalloc/internal/decay.h

```diff
@@ -146,4 +146,14 @@ void decay_reinit(decay_t *decay, nstime_t *cur_time, ssize_t decay_ms);
 bool decay_maybe_advance_epoch(decay_t *decay, nstime_t *new_time,
     size_t current_npages);
 
+typedef struct purge_interval_opts_s purge_interval_opts_t;
+struct purge_interval_opts_s {
+	uint64_t min_interval_ns;
```

Comment:
I think that `min_interval_ns` probably belongs in the background thread code, since it's not really fundamental to the decay logic, but rather is just to avoid short background thread sleeps. A little cleaner would be to have the decay module *just* return the number of nanos there will be before there are pages that should be purged.

Reply:
Maybe it logically belongs in the background thread, but its removal affects the binary search in `arena_decay_compute_purge_interval_impl`. Lower bound of the binary search is calculated based on `min_interval_ns`. We could always assign lower bound to 2 though, but at the cost of additional binary search iterations. Is it worth it?

Reply:
Hmm, that's a good point. My inclination is that the extra computation is fine if it gets us a slightly nicer API (this code isn't hot at all), but I don't feel so strongly about it that I'll argue about it if you disagree.